BALSAMIC/utils/models.py

```diff
@@ -688,3 +691,77 @@ class BalsamicWorkflowConfig(BaseModel):
     umiextract: UMIParamsUMIextract
     umiconsensuscall: UMIParamsConsensuscall
     tnscope_umi: UMIParamsTNscope
+
+
+class QCConditionModel(BaseModel):
+    """Defines the quality control filtering parameters
+
+    Attributes:
+        norm: string (optional); validation condition
+        threshold: float (optional); validation cut off
+    """
+
+    norm: str
+    threshold: float
+
+    @validator("norm")
+    def check_operator(cls, value) -> str:
+        """Validates if the retrieved norm is a valid operator"""
+        assert value in [
+            "lt",
+            "le",
+            "eq",
+            "ne",
+            "ge",
+            "gt",
+        ], f"{value} is not a valid condition for QC filtering"
+        return value
+
+
+class QCMetricsModel(BaseModel):
+    """Defines the quality control metrics model
+
+    Attributes:
+        name: str (required); quality control metric name
+        value: float (required); metrics value
+        condition: QCConditionModel (optional); condition for QC validation
+    """
+
+    name: str
+    value: float
+    condition: Optional[QCConditionModel]
+
+
+class QCCheckModel(BaseModel):
+    """Defines the quality control validation model
+
+    Attributes:
+        metrics: Dict(sample_name, QCMetricsModel) (required); quality control metric attributes
+    """
+
+    metrics: Dict[str, List[QCMetricsModel]]
+
+    @validator("metrics", each_item=True)
+    def check_metric(cls, value):
+        """Checks if each metric meets the filtering conditions"""
+        for metric in value:
+            if metric.condition is not None and not eval(
+                f"operator.{metric.condition.norm}({metric.value},{metric.condition.threshold})"
+            ):
```

Comment:
Use of `eval` almost always makes debugging troublesome.

condition is extracted, then the operator is passed to `operator` to evaluate the metric.

What you can do is to use following:

![image](https://user-images.githubusercontent.com/1086877/134899453-79114610-ef84-45e4-af45-1d87a9594e74.png)

```python
import operator
valid_ops={
"eq": operator.eq,
"lt": operator.lt,
}

fold80_threshold=1.8;
actual_fold80=1.6;

metric_conditon="eq"
valid_ops[metric_conditon](actual_fold80, fold80_threshold)

new_metric_condition="lt"
valid_ops[new_metric_condition](actual_fold80, fold80_threshold)
```


Also, i suggest to `continue` if `metric.conditon` is `None`:

```suggestion
            if metric.condition is None:
                 continue
```